exonum/src/encoding/protobuf/mod.rs

```diff
@@ -53,17 +56,17 @@ pub trait ProtobufConvert: Sized {
 
 impl<T> BinaryForm for T
 where
-    T: ProtobufConvert,
-    <T as ProtobufConvert>::ProtoStruct: Message,
+    T: Message,
 {
     fn encode(&self) -> Result<Vec<u8>, Error> {
-        Ok(self.to_pb().write_to_bytes().unwrap())
+        Ok(self.write_to_bytes().unwrap())
```

Comment:
I guess It will be better, won't it ? 
```suggestion
        self.write_to_bytes().map_err(|e| Error::Other(Box::new(e)))
```